src/diskio/mod.rs

```diff
@@ -452,7 +453,7 @@ pub(crate) fn get_executor<'a>(
 ) -> Result<Box<dyn Executor + 'a>> {
     // If this gets lots of use, consider exposing via the config file.
     let thread_count = match process().var("RUSTUP_IO_THREADS") {
-        Err(_) => num_cpus::get(),
+        Err(_) => available_parallelism()?.get(),
```

Comment:
```suggestion
        Err(_) => available_parallelism().map(|p|p.get()).unwrap_or(1),
```

or something similar - the docs for available parallelism document failure cases where it is better for rustup to run slower than to fail.

Reply:
Sounds good!